utils/build.ps1

```diff
@@ -995,18 +995,21 @@ function Build-CMakeProject {
       }
     }
 
-    if ($UseMSVCCompilers.Contains("C")) {
-      TryAdd-KeyValue $Defines CMAKE_C_COMPILER cl
-      if ($EnableCaching) {
+    if ($EnableCaching) {
+      if ($UseMSVCCompilers.Contains("C") -Or $UsePinnedCompilers.Contains("C")) {
```

Comment:
Can we drop this check entirely? That is, simplify to:

```ps1
if ($EnableCaching) {
  TryAdd-KeyValue $Defines CMAKE_C_COMPILER sccache
  TryAdd-KeyValue $Defines CMAKE_CXX_COMPILER sccache
}
```

Reply:
Right, should be no problem to have both properties even if we don't use both compilers. Done.